internal/erasurecoding/erasurecoding_test.go

```diff
@@ -0,0 +1,180 @@
+package erasurecoding
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/eigerco/strawberry/internal/common"
+	"github.com/eigerco/strawberry/internal/testutils"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/rand"
+)
+
+func TestUnzipLace(t *testing.T) {
+	testCases := []struct {
+		name string
+		data []byte
+		n    int
+		want []byte
+	}{
+		{
+			name: "perfectlyDivsible",
+			data: []byte{1, 2, 3, 4, 5, 6},
+			n:    2,
+			want: []byte{1, 2, 3, 4, 5, 6},
+		},
+		{
+			name: "withPadding",
+			data: []byte{1, 2, 3, 4, 5},
+			n:    2,
+			want: []byte{1, 2, 3, 4, 5},
+		},
+		{
+			name: "singles",
+			data: []byte{1, 2, 3},
+			n:    3,
+			want: []byte{1, 2, 3},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			chunks := unzip(tc.n, tc.data)
+			data := lace(chunks, len(tc.data))
+
+			require.Equal(t, data, tc.want)
+		})
+	}
+}
+
+func TestEncodeDecodeRoundTripRandom(t *testing.T) {
+	dataSizes := []int{
+		1,
+		684,
+		1368,
+		2052,
+		4096,
+		4104,
+	}
+
+	seed := uint64(time.Now().UnixNano())
+	t.Logf("using random seed: %d", seed)
+	rng := rand.New(rand.NewSource(seed))
+
+	for _, size := range dataSizes {
+		t.Run(fmt.Sprintf("size%d", size), func(t *testing.T) {
+			// Generate some random data.
+			data := make([]byte, size)
+			_, err := rng.Read(data)
+			require.NoError(t, err)
+
+			// Encode the data
+			shards, err := Encode(data)
+			require.NoError(t, err)
+			require.Equal(t, len(shards), OriginalShards+RecoveryShards)
+
+			// Randomly remove RecoveryShards shards.
+			indices := rng.Perm(RecoveryShards)
+			for i := range indices {
+				shards[i] = nil
+			}
+
+			// Decode the data with the remaining shards.
+			dataOut, err := Decode(shards, size)
+			if err != nil {
+				t.Fatalf("Decode failed: %v", err)
+			}
+
+			require.Equal(t, data, dataOut)
+		})
+	}
+}
+
+func TestEncodeDecodeInitialTestVector(t *testing.T) {
```

Comment:
Should we move this test to the integration folder to be consistent with the rest that tests test vectors? Also saw that we have the `readsolomon_test` also testing vectors.

Reply:
Do you feel like this is really an integration test if it's only testing one component?

We also only have one semi official test vector at the moment, maybe that can be moved to integration when the proper erasure coding vectors land?

Reply:
I could have probably just inlined that initial test vector

Reply:
Imo if it is a test vector it should live with the other test vector cases. If you inline it you hardcode the test vector, right? It's called integration test, but that is just an incorrect naming. In reality it is a test vector folder.

Reply:
This isn't an official test vector technically, I'll remove the vectors folder and just put the json in the package. When the official vectors land then I'll add those tests in the integration tests folder.

Reply:
(basically I also want a test like this at the package level)